packages/astro/src/@types/astro.ts

```diff
@@ -1272,6 +1272,27 @@ export interface AstroUserConfig {
 		 * ```
 		 */
 		viewTransitions?: boolean;
+
+		/**
+		 * @docs
+		 * @name experimental.optimizeHoistedScript
+		 * @type {boolean}
+		 * @default `false`
+		 * @version 2.10.4
+		 * @description
+		 * Enable hoisted script analysis optimization to prevent unused components' script from being included in a page unexpectedly.
+		 * The optimization is best-effort and may inversely miss including the used scripts. Make sure to double-check your built pages
+		 * before publishing.
```

Comment:
```suggestion
		 * Prevents unused components' scripts from being included in a page unexpectedly.
		 * The optimization is best-effort and may inversely miss including the used scripts. Make sure to double-check your built pages
		 * before publishing.
		 * Enable hoisted script analysis optimization by adding the experimental flag:
```
Nice, @bluwy!  For future proofing, since presumably at some point this will no longer be experimental, we could maybe just start with the description of what it does, and move the instruction to enable closer to the code block? That way, this wording will continue to work and can be reused if the config option description is needed in the future. :raised_hands: !

Reply:
Sounds great to me. I was following the description pattern in the other experimental options for this, but either way works for me.